magellan-library/src/main/java/com/wealthfront/magellan/navigation/LinearNavigator.kt

```diff
@@ -16,37 +16,31 @@ import com.wealthfront.magellan.lifecycle.LifecycleState.Created
 import com.wealthfront.magellan.lifecycle.LifecycleState.Destroyed
 import com.wealthfront.magellan.lifecycle.LifecycleState.Resumed
 import com.wealthfront.magellan.lifecycle.LifecycleState.Shown
-import com.wealthfront.magellan.lifecycle.LifecycleStateMachine
 import com.wealthfront.magellan.transitions.DefaultTransition
 import com.wealthfront.magellan.view.whenMeasured
-import java.util.Deque
-import java.util.LinkedList
+import java.util.Stack
 
 class LinearNavigator(
   private val container: () -> ScreenContainer
 ) : Navigator, LifecycleAwareComponent() {
 
-  private val lifecycleStateMachine = LifecycleStateMachine()
+  private var containerView: ScreenContainer? = null
 
   @VisibleForTesting
-  override val backStack: LinkedList<NavigationEvent> = LinkedList()
-
-  private var ghostView: View? = null
-  private var containerView: ScreenContainer? = null
+  override val backStack: Stack<NavigationEvent> = Stack()
```

Comment:
Using a stack now so we need to do the empty check but also restrict the api a lot which is good here.